### ISSUE TITLE
fix: `mason update` doesn't work

### ISSUE THREAD

**Issue by SAGARSURI** [felangel/mason#727]
**Description**

When I run the command `mason update`. It's not updating mason to the latest version.

**Steps To Reproduce**

- Check the installed version of the mason.
- Run `mason update`
- Run `mason --version`
- mason is not updated

**Expected Behavior**

mason should be updated to the latest version

**Screenshots**

<img width="598" alt="Screenshot 2023-02-02 at 11 40 50" src="https://user-images.githubusercontent.com/14856659/216302967-9ccd75b2-93dc-4d79-b4b2-07f0c62f2413.png">

**Additional Context**

**Comment by SAGARSURI** [comment]
So the issue with `mason update` is, it doesn't work if a user has installed mason using `brew`.

**Comment by felangel** [comment]
> So the issue with `mason update` is, it doesn't work if a user has installed mason using `brew`.
> I am planning to create a package(`brew_updater`) similar to `pub_updater` but that will use brew to update an existing dependency. What do you think?

Good catch! Yeah that seems reasonable. We’ll also need to have logic to determine whether the mason installation was done through pub or brew in order to determine which mechanism to use for update checks.

**Comment by SAGARSURI** [comment]
Perfect! I will create a PR soon 😃 . Thanks for your input.

**Comment by luongvo** [comment]
+1 does not work for me
